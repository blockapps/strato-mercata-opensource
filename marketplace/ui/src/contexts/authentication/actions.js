import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  check: 'auth/check',
  checkSuccessful: 'auth/check_successful',
  checkFailed: 'auth/check_failed',
  checkFailedInternalServerError: 'auth/check_failed_internal_server_error',
  fetchUsers: 'FETCH_USERS',
  fetchUsersSuccessful: 'FETCH_USERS_SUCCESSFUL',
  fetchUsersFailed: 'FETCH_USERS_FAILED',
  logoutCheck: 'auth/logout_check',
  logoutSuccessful: 'auth/logout_success',
  logoutFailed: 'auth/logout_failed',
};

const actions = {
  check: async (dispatch) => {
    dispatch({ type: actionDescriptors.check });
    try {
      let response = await fetch(`${apiUrl}/users/me`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });
      const body = await response.json();
      if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        {
          dispatch({
            type: actionDescriptors.checkFailedInternalServerError,
            error: body.error,
          });
          return;
        }
      } else if (
        response.status === RestStatus.UNAUTHORIZED ||
        response.status === RestStatus.FORBIDDEN
      ) {
        dispatch({
          type: actionDescriptors.checkFailed,
          payload: body.error.loginUrl,
        });
        return;
      } else if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.checkSuccessful,
          payload: { ...body.data },
        });
        return;
      }
      dispatch({ type: actionDescriptors.checkFailed, payload: undefined });
    } catch (err) {
      dispatch({ type: actionDescriptors.checkFailed, payload: undefined });
    }
  },
  fetchUsers: async (dispatch) => {
    dispatch({ type: actionDescriptors.fetchUsers });
    try {
      const response = await fetch(`${apiUrl}/users`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchUsersSuccessful,
          payload: body.data,
        });
        return;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.fetchUsersFailed,
          error: 'Unauthorized while fetching users',
        });
        window.location.href = body.error.loginUrl;
      }
      dispatch({
        type: actionDescriptors.fetchUsersFailed,
        payload: undefined,
      });
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchUsersFailed,
        payload: undefined,
      });
    }
  },
  logout: async (dispatch) => {
    dispatch({ type: actionDescriptors.logoutCheck });
    try {
      const response = await fetch(`${apiUrl}/authentication/logout`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });
      const body = await response.json();
      if (response.status === RestStatus.OK) {
        window.location.href = body.data.logoutUrl;
        localStorage.removeItem('loginCount');
        dispatch({
          type: actionDescriptors.logoutSuccessful,
          payload: body.data.logoutUrl,
        });
        return;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.logoutFailed,
          error: 'Unauthorized while logging out',
        });
        window.location.href = body.error.loginUrl;
      }
      dispatch({ type: actionDescriptors.logoutFailed, payload: undefined });
    } catch (err) {
      dispatch({ type: actionDescriptors.logoutFailed, payload: undefined });
    }
  },
};

export { actionDescriptors, actions };
