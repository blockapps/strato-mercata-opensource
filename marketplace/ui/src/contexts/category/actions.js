import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  createCategory: 'create_category',
  createCategorySuccessful: 'create_category_successful',
  createCategoryFailed: 'create_category_failed',
  fetchCategory: 'fetch_categorys',
  fetchCategorySuccessful: 'fetch_category_successful',
  fetchCategoryFailed: 'fetch_category_failed',
  resetMessage: 'reset_message',
  setMessage: 'set_message',
};

const actions = {
  resetMessage: (dispatch) => {
    dispatch({ type: actionDescriptors.resetMessage });
  },

  setMessage: (dispatch, message, success = false) => {
    dispatch({ type: actionDescriptors.setMessage, message, success });
  },

  createCategory: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.createCategory });

    try {
      const response = await fetch(`${apiUrl}/category`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.createCategorySuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Category created successfully', true);
        return true;
      }

      dispatch({
        type: actionDescriptors.createCategoryFailed,
        error: 'Error while creating Category',
      });
      actions.setMessage(dispatch, 'Error while creating Category');
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.createCategoryFailed,
        error: 'Error while creating Category',
      });
      actions.setMessage(dispatch, 'Error while creating Category');
    }
  },

  fetchCategories: async (dispatch) => {
    dispatch({ type: actionDescriptors.fetchCategory });

    try {
      const response = await fetch(`${apiUrl}/category`, {
        method: HTTP_METHODS.GET,
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchCategorySuccessful,
          payload: body.data,
        });
        return;
      } else if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        dispatch({
          type: actionDescriptors.fetchCategoryFailed,
          error: 'Error while fetching category',
        });
        return false;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.fetchCategoryFailed,
          error: 'Unauthorized while fetching categories',
        });
        window.location.href = body.error.loginUrl;
      }

      dispatch({
        type: actionDescriptors.fetchCategoryFailed,
        error: body.error,
      });
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchCategoryFailed,
        error: 'Error while fetching category',
      });
    }
  },
};

export { actionDescriptors, actions };
