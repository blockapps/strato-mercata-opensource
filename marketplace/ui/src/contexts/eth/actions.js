import RestStatus from 'http-status-codes';
import { apiUrl, HTTP_METHODS } from '../../helpers/constants';

const actionDescriptors = {
  resetMessage: 'reset_message',
  setMessage: 'set_message',
  fetchETHSTAddress: 'fetch_ethst_address',
  fetchETHSTAddressSuccessful: 'fetch_ethst_address_successful',
  fetchETHSTAddressFailed: 'fetch_ethst_address_failed',
  addHash: 'add_hash',
  addHashSuccessful: 'add_hash_successful',
  addHashFailed: 'add_hash_failed',
};

const actions = {
  resetMessage: (dispatch) => {
    dispatch({ type: actionDescriptors.resetMessage });
  },

  setMessage: (dispatch, message, success = false) => {
    dispatch({ type: actionDescriptors.setMessage, message, success });
  },

  fetchETHSTAddress: async (dispatch) => {
    dispatch({ type: actionDescriptors.fetchETHSTAddress });
    try {
      let response = await fetch(`${apiUrl}/tokens/address`, {
        method: HTTP_METHODS.GET,
        credentials: 'same-origin',
      });

      const body = await response.json();
      if (
        response.status === RestStatus.UNAUTHORIZED ||
        response.status === RestStatus.FORBIDDEN
      ) {
        dispatch({
          type: actionDescriptors.fetchETHSTAddressFailed,
          payload: 'Error while fetching ETHST address',
        });
        return null;
      }

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.fetchETHSTAddressSuccessful,
          payload: body?.data,
        });
        return body.data;
      }

      dispatch({
        type: actionDescriptors.fetchETHSTAddressFailed,
        payload: 'Error while fetching ETHST address',
      });
      return null;
    } catch (err) {
      dispatch({
        type: actionDescriptors.fetchETHSTAddressFailed,
        payload: 'Error while fetching ETHST address',
      });
      return null;
    }
  },

  addHash: async (dispatch, payload) => {
    dispatch({ type: actionDescriptors.addHash });

    try {
      const response = await fetch(`${apiUrl}/tokens/addHash`, {
        method: HTTP_METHODS.POST,
        credentials: 'same-origin',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      const body = await response.json();

      if (response.status === RestStatus.OK) {
        dispatch({
          type: actionDescriptors.addHashSuccessful,
          payload: body.data,
        });
        actions.setMessage(dispatch, 'Hash added successfully', true);
        return true;
      } else if (response.status === RestStatus.CONFLICT) {
        dispatch({
          type: actionDescriptors.addHashFailed,
          error: body.error.message,
        });
        actions.setMessage(dispatch, body.error.message);
        return false;
      } else if (response.status === RestStatus.INTERNAL_SERVER_ERROR) {
        dispatch({
          type: actionDescriptors.addHashFailed,
          error: 'Error while adding Hash',
        });
        actions.setMessage(dispatch, 'Error while adding Hash');
        return false;
      } else if (response.status === RestStatus.UNAUTHORIZED) {
        dispatch({
          type: actionDescriptors.addHashFailed,
          error: 'Unauthorized while adding Hash',
        });
        window.location.href = body.error.loginUrl;
      }

      dispatch({
        type: actionDescriptors.addHashFailed,
        error: body.error,
      });
      actions.setMessage(dispatch, body.error);
      return false;
    } catch (err) {
      dispatch({
        type: actionDescriptors.addHashFailed,
        error: 'Error while adding Hash',
      });
      actions.setMessage(dispatch, 'Error while adding Hash');
    }
  },
};

export { actionDescriptors, actions };
