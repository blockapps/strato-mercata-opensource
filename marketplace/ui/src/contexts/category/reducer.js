import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createCategory:
      return {
        ...state,
        isCreateCategorySubmitting: true,
      };
    case actionDescriptors.createCategorySuccessful:
      return {
        ...state,
        category: action.payload,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.createCategoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.fetchCategory:
      return {
        ...state,
        iscategorysLoading: true,
      };
    case actionDescriptors.fetchCategorySuccessful:
      return {
        ...state,
        categorys: action.payload,
        iscategorysLoading: false,
      };
    case actionDescriptors.fetchCategoryFailed:
      return {
        ...state,
        error: action.error,
        iscategorysLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
