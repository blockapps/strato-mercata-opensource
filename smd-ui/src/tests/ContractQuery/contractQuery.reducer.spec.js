import reducer from '../../components/ContractQuery/contractQuery.reducer';
import { clearQueryString, queryCirrusSuccess, queryCirrusFailure, queryCirrusVarsSuccess, addQueryFilter, removeQueryFilter } from '../../components/ContractQuery/contractQuery.actions';
import { queryCirrusMock, error, queryCirrusVarsMock } from './contractQueryMock';

describe('ContractQuery: reducer', () => {

  // INITIAL_STATE
  test('set initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  // CLEAR_QUERY_STRING
  test('clear query', () => {
    const action = clearQueryString();
    const initialState = {
      queryString: 'state=eq.1',
      queryResults: null,
      tags: [],
      vars: null,
      error: null
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  describe('fetch querycirrus', () => {
    // QUERY_CIRRUS_SUCCESS
    test('success', () => {
      const action = queryCirrusSuccess(queryCirrusMock);
      const initialState = {
        queryString: '',
        queryResults: null,
        tags: [],
        vars: null,
        error: null
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // QUERY_CIRRUS_FAILURE
    test('failure', () => {
      const action = queryCirrusFailure(error);
      const initialState = {
        queryString: '',
        queryResults: null,
        tags: [],
        vars: null,
        error: null
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  })

  // QUERY_CIRRUS_VARS_SUCCESS
  test('queryCirrusVars success', () => {
    const action = queryCirrusVarsSuccess(queryCirrusVarsMock.xabi.vars);
    const initialState = {
      queryString: '',
      queryResults: queryCirrusMock,
      tags: [],
      vars: null,
      error: null
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  // ADD_QUERY_FILTER
  test('add new filter', () => {
    const payload = {
      field: 'state',
      operator: 'eq',
      value: 1
    };
    const action = addQueryFilter(payload.field, payload.operator, payload.value);
    const initialState = {
      queryString: '',
      queryResults: queryCirrusMock,
      tags: [],
      vars: queryCirrusVarsMock.xabi.vars,
      error: null
    }
    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  // REMOVE_QUERY_FILTER with it's cases
  describe('remove query filter', () => {

    test('that contains only one tag', () => {
      const action = removeQueryFilter(0);
      const initialState = {
        queryString: 'state=eq.1',
        queryResults: queryCirrusMock,
        tags: [{
          display: "state eq 1",
          field: "state",
          operator: "eq",
          value: 1,
        }],
        vars: queryCirrusVarsMock.xabi.vars,
        error: null
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    test('that contains two tags', () => {
      const action = removeQueryFilter(0);
      const initialState = {
        queryString: 'state=eq.1&amount=eq.5678',
        queryResults: queryCirrusMock,
        tags: [
          {
            field: "state",
            operator: "eq",
            value: "1",
            display: "state eq 1"
          },
          {
            field: "amount",
            operator: "eq",
            value: "5678",
            display: "amount eq 5678"
          }
        ],
        vars: queryCirrusVarsMock.xabi.vars,
        error: null
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    test('center tags from multiple tags', () => {
      const action = removeQueryFilter(1);
      const initialState = {
        queryString: 'state=eq.1&amount=eq.5678&name=eq.P',
        queryResults: queryCirrusMock,
        tags: [
          {
            field: "state",
            operator: "eq",
            value: "1",
            display: "state eq 1"
          },
          {
            field: "amount",
            operator: "eq",
            value: "5678",
            display: "amount eq 5678"
          },
          {
            display: "name eq P",
            field: "name",
            operator: "eq",
            value: "P"
          }
        ],
        vars: queryCirrusVarsMock.xabi.vars,
        error: null
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  });

});
