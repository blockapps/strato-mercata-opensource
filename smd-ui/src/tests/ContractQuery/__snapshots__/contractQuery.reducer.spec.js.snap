// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractQuery: reducer add new filter 1`] = `
Object {
  "error": null,
  "queryResults": Array [
    Object {
      "address": "ddbfe1fe25dfbb91633c576aaadd8d3e377be47b",
      "amount": 5678,
      "id": 149,
      "name": "P_ ?%#@!:* 41769",
      "state": "1",
      "supplier": "Supplier1_1301_24669",
    },
    Object {
      "address": "5112a195a0bf46dbeac6ce2e9a2fbbed50dc3e44",
      "amount": 67,
      "id": 150,
      "name": "P_ ?%#@!:* 92348",
      "state": "2",
      "supplier": "Supplier1",
    },
    Object {
      "address": "eff065f2e1cf483daee729f1f45c92db46fe1092",
      "amount": 1067,
      "id": 151,
      "name": "P_ ?%#@!:* 44366",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "bd0a5c9f7965e3ff970b2b69eb686a9dd6d93ea3",
      "amount": 32,
      "id": 152,
      "name": "P_ ?%#@!:* 46441",
      "state": "2",
      "supplier": "Supplier0_1301_46441",
    },
    Object {
      "address": "2e126696dd113e8ef1a96f51d4fe21bdec5bb8d4",
      "amount": 32,
      "id": 153,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier1_1301_46441",
    },
    Object {
      "address": "421022d27ad192eb2979d0d102945d05b1b62186",
      "amount": 32,
      "id": 154,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier2_1301_46441",
    },
    Object {
      "address": "1bf900ac2d58688ebd9847ec9993533b4207165e",
      "amount": 5678,
      "id": 155,
      "name": "P_ ?%#@!:* 80772",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "ad601648712a607acf73cae058d4bc8601c1a724",
      "amount": 100,
      "id": 3,
      "name": "Project_ ? 1452_77996",
      "state": "2",
      "supplier": "Supplier1",
    },
  ],
  "queryString": "state=eq.1",
  "tags": Array [
    Object {
      "display": "state eq 1",
      "field": "state",
      "operator": "eq",
      "value": 1,
    },
  ],
  "vars": Object {
    "amount": Object {
      "atBytes": 96,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "id": Object {
      "atBytes": 0,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "name": Object {
      "atBytes": 32,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
    "state": Object {
      "atBytes": 128,
      "contents": "BidState",
      "public": true,
      "type": "Label",
    },
    "supplier": Object {
      "atBytes": 64,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
  },
}
`;

exports[`ContractQuery: reducer clear query 1`] = `
Object {
  "error": null,
  "queryResults": null,
  "queryString": "",
  "tags": Array [],
  "vars": null,
}
`;

exports[`ContractQuery: reducer fetch querycirrus failure 1`] = `
Object {
  "error": "ERROR",
  "queryResults": null,
  "queryString": "",
  "tags": Array [],
  "vars": null,
}
`;

exports[`ContractQuery: reducer fetch querycirrus success 1`] = `
Object {
  "error": null,
  "queryResults": Array [
    Object {
      "address": "ddbfe1fe25dfbb91633c576aaadd8d3e377be47b",
      "amount": 5678,
      "id": 149,
      "name": "P_ ?%#@!:* 41769",
      "state": "1",
      "supplier": "Supplier1_1301_24669",
    },
    Object {
      "address": "5112a195a0bf46dbeac6ce2e9a2fbbed50dc3e44",
      "amount": 67,
      "id": 150,
      "name": "P_ ?%#@!:* 92348",
      "state": "2",
      "supplier": "Supplier1",
    },
    Object {
      "address": "eff065f2e1cf483daee729f1f45c92db46fe1092",
      "amount": 1067,
      "id": 151,
      "name": "P_ ?%#@!:* 44366",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "bd0a5c9f7965e3ff970b2b69eb686a9dd6d93ea3",
      "amount": 32,
      "id": 152,
      "name": "P_ ?%#@!:* 46441",
      "state": "2",
      "supplier": "Supplier0_1301_46441",
    },
    Object {
      "address": "2e126696dd113e8ef1a96f51d4fe21bdec5bb8d4",
      "amount": 32,
      "id": 153,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier1_1301_46441",
    },
    Object {
      "address": "421022d27ad192eb2979d0d102945d05b1b62186",
      "amount": 32,
      "id": 154,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier2_1301_46441",
    },
    Object {
      "address": "1bf900ac2d58688ebd9847ec9993533b4207165e",
      "amount": 5678,
      "id": 155,
      "name": "P_ ?%#@!:* 80772",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "ad601648712a607acf73cae058d4bc8601c1a724",
      "amount": 100,
      "id": 3,
      "name": "Project_ ? 1452_77996",
      "state": "2",
      "supplier": "Supplier1",
    },
  ],
  "queryString": "",
  "tags": Array [],
  "vars": null,
}
`;

exports[`ContractQuery: reducer queryCirrusVars success 1`] = `
Object {
  "error": null,
  "queryResults": Array [
    Object {
      "address": "ddbfe1fe25dfbb91633c576aaadd8d3e377be47b",
      "amount": 5678,
      "id": 149,
      "name": "P_ ?%#@!:* 41769",
      "state": "1",
      "supplier": "Supplier1_1301_24669",
    },
    Object {
      "address": "5112a195a0bf46dbeac6ce2e9a2fbbed50dc3e44",
      "amount": 67,
      "id": 150,
      "name": "P_ ?%#@!:* 92348",
      "state": "2",
      "supplier": "Supplier1",
    },
    Object {
      "address": "eff065f2e1cf483daee729f1f45c92db46fe1092",
      "amount": 1067,
      "id": 151,
      "name": "P_ ?%#@!:* 44366",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "bd0a5c9f7965e3ff970b2b69eb686a9dd6d93ea3",
      "amount": 32,
      "id": 152,
      "name": "P_ ?%#@!:* 46441",
      "state": "2",
      "supplier": "Supplier0_1301_46441",
    },
    Object {
      "address": "2e126696dd113e8ef1a96f51d4fe21bdec5bb8d4",
      "amount": 32,
      "id": 153,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier1_1301_46441",
    },
    Object {
      "address": "421022d27ad192eb2979d0d102945d05b1b62186",
      "amount": 32,
      "id": 154,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier2_1301_46441",
    },
    Object {
      "address": "1bf900ac2d58688ebd9847ec9993533b4207165e",
      "amount": 5678,
      "id": 155,
      "name": "P_ ?%#@!:* 80772",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "ad601648712a607acf73cae058d4bc8601c1a724",
      "amount": 100,
      "id": 3,
      "name": "Project_ ? 1452_77996",
      "state": "2",
      "supplier": "Supplier1",
    },
  ],
  "queryString": "",
  "tags": Array [],
  "vars": Object {
    "amount": Object {
      "atBytes": 96,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "id": Object {
      "atBytes": 0,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "name": Object {
      "atBytes": 32,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
    "state": Object {
      "atBytes": 128,
      "contents": "BidState",
      "public": true,
      "type": "Label",
    },
    "supplier": Object {
      "atBytes": 64,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
  },
}
`;

exports[`ContractQuery: reducer remove query filter center tags from multiple tags 1`] = `
Object {
  "error": null,
  "queryResults": Array [
    Object {
      "address": "ddbfe1fe25dfbb91633c576aaadd8d3e377be47b",
      "amount": 5678,
      "id": 149,
      "name": "P_ ?%#@!:* 41769",
      "state": "1",
      "supplier": "Supplier1_1301_24669",
    },
    Object {
      "address": "5112a195a0bf46dbeac6ce2e9a2fbbed50dc3e44",
      "amount": 67,
      "id": 150,
      "name": "P_ ?%#@!:* 92348",
      "state": "2",
      "supplier": "Supplier1",
    },
    Object {
      "address": "eff065f2e1cf483daee729f1f45c92db46fe1092",
      "amount": 1067,
      "id": 151,
      "name": "P_ ?%#@!:* 44366",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "bd0a5c9f7965e3ff970b2b69eb686a9dd6d93ea3",
      "amount": 32,
      "id": 152,
      "name": "P_ ?%#@!:* 46441",
      "state": "2",
      "supplier": "Supplier0_1301_46441",
    },
    Object {
      "address": "2e126696dd113e8ef1a96f51d4fe21bdec5bb8d4",
      "amount": 32,
      "id": 153,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier1_1301_46441",
    },
    Object {
      "address": "421022d27ad192eb2979d0d102945d05b1b62186",
      "amount": 32,
      "id": 154,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier2_1301_46441",
    },
    Object {
      "address": "1bf900ac2d58688ebd9847ec9993533b4207165e",
      "amount": 5678,
      "id": 155,
      "name": "P_ ?%#@!:* 80772",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "ad601648712a607acf73cae058d4bc8601c1a724",
      "amount": 100,
      "id": 3,
      "name": "Project_ ? 1452_77996",
      "state": "2",
      "supplier": "Supplier1",
    },
  ],
  "queryString": "state=eq.1&name=eq.P",
  "tags": Array [
    Object {
      "display": "state eq 1",
      "field": "state",
      "operator": "eq",
      "value": "1",
    },
    Object {
      "display": "name eq P",
      "field": "name",
      "operator": "eq",
      "value": "P",
    },
  ],
  "vars": Object {
    "amount": Object {
      "atBytes": 96,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "id": Object {
      "atBytes": 0,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "name": Object {
      "atBytes": 32,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
    "state": Object {
      "atBytes": 128,
      "contents": "BidState",
      "public": true,
      "type": "Label",
    },
    "supplier": Object {
      "atBytes": 64,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
  },
}
`;

exports[`ContractQuery: reducer remove query filter that contains only one tag 1`] = `
Object {
  "error": null,
  "queryResults": Array [
    Object {
      "address": "ddbfe1fe25dfbb91633c576aaadd8d3e377be47b",
      "amount": 5678,
      "id": 149,
      "name": "P_ ?%#@!:* 41769",
      "state": "1",
      "supplier": "Supplier1_1301_24669",
    },
    Object {
      "address": "5112a195a0bf46dbeac6ce2e9a2fbbed50dc3e44",
      "amount": 67,
      "id": 150,
      "name": "P_ ?%#@!:* 92348",
      "state": "2",
      "supplier": "Supplier1",
    },
    Object {
      "address": "eff065f2e1cf483daee729f1f45c92db46fe1092",
      "amount": 1067,
      "id": 151,
      "name": "P_ ?%#@!:* 44366",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "bd0a5c9f7965e3ff970b2b69eb686a9dd6d93ea3",
      "amount": 32,
      "id": 152,
      "name": "P_ ?%#@!:* 46441",
      "state": "2",
      "supplier": "Supplier0_1301_46441",
    },
    Object {
      "address": "2e126696dd113e8ef1a96f51d4fe21bdec5bb8d4",
      "amount": 32,
      "id": 153,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier1_1301_46441",
    },
    Object {
      "address": "421022d27ad192eb2979d0d102945d05b1b62186",
      "amount": 32,
      "id": 154,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier2_1301_46441",
    },
    Object {
      "address": "1bf900ac2d58688ebd9847ec9993533b4207165e",
      "amount": 5678,
      "id": 155,
      "name": "P_ ?%#@!:* 80772",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "ad601648712a607acf73cae058d4bc8601c1a724",
      "amount": 100,
      "id": 3,
      "name": "Project_ ? 1452_77996",
      "state": "2",
      "supplier": "Supplier1",
    },
  ],
  "queryString": "",
  "tags": Array [],
  "vars": Object {
    "amount": Object {
      "atBytes": 96,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "id": Object {
      "atBytes": 0,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "name": Object {
      "atBytes": 32,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
    "state": Object {
      "atBytes": 128,
      "contents": "BidState",
      "public": true,
      "type": "Label",
    },
    "supplier": Object {
      "atBytes": 64,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
  },
}
`;

exports[`ContractQuery: reducer remove query filter that contains two tags 1`] = `
Object {
  "error": null,
  "queryResults": Array [
    Object {
      "address": "ddbfe1fe25dfbb91633c576aaadd8d3e377be47b",
      "amount": 5678,
      "id": 149,
      "name": "P_ ?%#@!:* 41769",
      "state": "1",
      "supplier": "Supplier1_1301_24669",
    },
    Object {
      "address": "5112a195a0bf46dbeac6ce2e9a2fbbed50dc3e44",
      "amount": 67,
      "id": 150,
      "name": "P_ ?%#@!:* 92348",
      "state": "2",
      "supplier": "Supplier1",
    },
    Object {
      "address": "eff065f2e1cf483daee729f1f45c92db46fe1092",
      "amount": 1067,
      "id": 151,
      "name": "P_ ?%#@!:* 44366",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "bd0a5c9f7965e3ff970b2b69eb686a9dd6d93ea3",
      "amount": 32,
      "id": 152,
      "name": "P_ ?%#@!:* 46441",
      "state": "2",
      "supplier": "Supplier0_1301_46441",
    },
    Object {
      "address": "2e126696dd113e8ef1a96f51d4fe21bdec5bb8d4",
      "amount": 32,
      "id": 153,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier1_1301_46441",
    },
    Object {
      "address": "421022d27ad192eb2979d0d102945d05b1b62186",
      "amount": 32,
      "id": 154,
      "name": "P_ ?%#@!:* 46441",
      "state": "3",
      "supplier": "Supplier2_1301_46441",
    },
    Object {
      "address": "1bf900ac2d58688ebd9847ec9993533b4207165e",
      "amount": 5678,
      "id": 155,
      "name": "P_ ?%#@!:* 80772",
      "state": "1",
      "supplier": "Supplier1",
    },
    Object {
      "address": "ad601648712a607acf73cae058d4bc8601c1a724",
      "amount": 100,
      "id": 3,
      "name": "Project_ ? 1452_77996",
      "state": "2",
      "supplier": "Supplier1",
    },
  ],
  "queryString": "amount=eq.5678",
  "tags": Array [
    Object {
      "display": "amount eq 5678",
      "field": "amount",
      "operator": "eq",
      "value": "5678",
    },
  ],
  "vars": Object {
    "amount": Object {
      "atBytes": 96,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "id": Object {
      "atBytes": 0,
      "bytes": 32,
      "public": true,
      "signed": false,
      "type": "Int",
    },
    "name": Object {
      "atBytes": 32,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
    "state": Object {
      "atBytes": 128,
      "contents": "BidState",
      "public": true,
      "type": "Label",
    },
    "supplier": Object {
      "atBytes": 64,
      "dynamic": true,
      "public": true,
      "type": "String",
    },
  },
}
`;

exports[`ContractQuery: reducer set initial state 1`] = `
Object {
  "contractAddress": null,
  "error": null,
  "queryResults": null,
  "queryString": "",
  "tags": Array [],
  "vars": null,
}
`;
